TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -106,11 +106,15 @@ private void Enter(Node node, CodeElement context = null, SymbolTable table = nu
         {
             node.SymbolTable = table ?? SyntaxTree.CurrentNode.SymbolTable;
             if (_IsInsideProcedure)
+            {
                 node.SetFlag(Node.Flag.InsideProcedure, true);      //Set flag to know that this node belongs a Procedure or Function
+                if (node is WorkingStorageSection || node is LocalStorageSection || node is LinkageSection || node is ProcedureDivision)
+                    node.SetFlag(Node.Flag.ForceGetGeneratedLines, true);
```

Comment:
Performance issue:
Please set the flag in method Startxxxx of each Node (eg StartWorkingStorageSection)
This will avoid to check if `node is WorkingStorageSection`